build_test.go

```diff
@@ -68,49 +61,106 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 				Entries: []packit.BuildpackPlanEntry{},
 			},
 			Layers: packit.Layers{Path: layersDir},
-		})
-		Expect(err).NotTo(HaveOccurred())
+		}
+	})
 
-		Expect(result).To(Equal(packit.BuildResult{
-			Plan: packit.BuildpackPlan{
-				Entries: nil,
-			},
-			Layers: nil,
-			Launch: packit.LaunchMetadata{
-				Processes: []packit.Process{
-					{
-						Type:    "web",
-						Command: "poetry run some-script",
-						Default: true,
-					},
-				},
-			},
-		}))
+	it.After(func() {
+		Expect(os.RemoveAll(layersDir)).To(Succeed())
+		Expect(os.RemoveAll(cnbDir)).To(Succeed())
+		Expect(os.RemoveAll(workingDir)).To(Succeed())
 	})
 
-	context("failure cases", func() {
-		context("when the pyproject.toml parser returns an error", func() {
-			it.Before(func() {
-				pyProjectParser.ParseCall.Returns.Error = fmt.Errorf("some error")
-			})
+	context("with BP_POETRY_RUN_TARGET not set", func() {
+		it("returns a result that sets the 'poetry run' launch command", func() {
+			result, err := build(buildContext)
+			Expect(err).NotTo(HaveOccurred())
 
-			it("returns the error", func() {
-				_, err := build(packit.BuildContext{
-					WorkingDir: workingDir,
-					CNBPath:    cnbDir,
-					Stack:      "some-stack",
-					BuildpackInfo: packit.BuildpackInfo{
-						Name:    "Some Buildpack",
-						Version: "some-version",
-					},
-					Plan: packit.BuildpackPlan{
-						Entries: []packit.BuildpackPlanEntry{},
+			Expect(result).To(Equal(packit.BuildResult{
+				Plan: packit.BuildpackPlan{
+					Entries: nil,
+				},
+				Layers: nil,
+				Launch: packit.LaunchMetadata{
+					Processes: []packit.Process{
+						{
+							Type:    "web",
+							Command: "poetry run some-script",
+							Default: true,
+						},
 					},
-					Layers: packit.Layers{Path: layersDir},
+				},
+			}))
+
+			Expect(buffer.String()).To(ContainLines(
+				ContainSubstring("Finding the poetry run target"),
+				ContainSubstring("Found pyproject.toml script=some-script"),
+				ContainSubstring("Assigning launch process"),
+				ContainSubstring("web: poetry run some-script"),
+			))
+		})
+
+		context("failure cases", func() {
```

Comment:
Can we group all the failure cases in one single case at the bottom, rather than nesting them inside other contexts? That way we can quickly and easily see which tests are for functionality vs error handling.

Reply:
Sure

Reply:
This is done, although the failure cases context structure will mimic the context structure of the non-failure tests. I think it's ok either way.

Reply:
yeah I agree that the structure will look similar. My goal with this request is to strive for consistency across all the buildpacks. We have a pattern of grouping all error cases at the bottom of unit tests. We should be consistent with that - if we don't like the pattern we should discuss the pattern holistically and make the change consistently.